packages/evm/src/utils/helpers.ts

```diff
@@ -132,10 +132,61 @@ export const getSubstrateRecipientAddressInBytes = (
  * @param padding - number to padd the data
  * @returns {string}
  */
-export const toHex = (covertThis: string | number | BigNumber, padding: number): string => {
+export const toHex = (
+  covertThis: string | number | BigNumber | bigint,
+  padding: number,
+): string => {
   const amount = covertThis instanceof BigNumber ? covertThis : BigNumber.from(covertThis);
   return utils.hexZeroPad(utils.hexlify(amount), padding);
 };
+/**
+ * JS types to 0x Hex string
+ * required to initiate contract
+ * calls on destination EVM chain
+ * @param {string | BigNumber | number | boolean | bigint} param
+ * @returns {`0x{string}`}
+ */
+function createGenericCallParameter(param: string | BigNumber | number | boolean | bigint): string {
+  const DEFAULT_PADDING = 32;
+  switch (typeof param) {
+    case 'boolean':
+      return toHex(Number(param), DEFAULT_PADDING).substring(2);
+    case 'bigint':
+    case 'number':
+    case 'string':
+      return toHex(param, DEFAULT_PADDING).substring(2);
+    case 'object':
+      if (param instanceof BigNumber) {
+        return toHex(param, DEFAULT_PADDING).substring(2);
+      }
+      throw new Error('Unsupported parameter type.');
+    case 'symbol':
+    case 'undefined':
+    case 'function':
+      throw new Error('Unsupported parameter type.');
+  }
+}
+/**
+ *
+ * @param params
```

Comment:
create a small description on the params. Also there is no return type on the typedocs